src/test/java/hudson/plugins/copyartifact/CopyArtifactTest.java

```diff
@@ -346,8 +346,15 @@ public void testCopyToSlave() throws Exception {
     @Test
     public void testParameters() throws Exception {
         FreeStyleProject other = createArtifactProject(),
-                         p = createProject("$PROJSRC", null, "$BASE/*.txt", "$TARGET/bar",
+                         p = createProject("${PROJSRC}", null, "${BASE}/*.txt", "${TARGET}/bar",
                                            false, false, false, true);
+        ParameterDefinition paramDef = new StringParameterDefinition("PROJSRC",other.getName(), "");
+        ParameterDefinition paramDef2 = new StringParameterDefinition("BASE","","");
+        ParameterDefinition paramDef3 = new StringParameterDefinition("TARGET","foo","");
+        ParametersDefinitionProperty paramsDef = new ParametersDefinitionProperty(paramDef, paramDef2,paramDef3);
+        p.addProperty(paramsDef);
+        CopyArtifactPermissionProperty cpp=new CopyArtifactPermissionProperty("/"+p.getName());
```

Comment:
@jglick Because of SECURITY-170 jobs can not use parameters that are not explicitly declared, so if you run that test against jenkins 1.651.3 (last LTS) or jenkins over 2.3 (the ones with that patch applied) you get the following error and the PROJSRC, BASE and TARGET properties does not get expanded, resulting in the test failing

{noformat}
WARNING: Skipped parameter `PROJSRC` as it is undefined on `test1`. Set `-Dhudson.model.ParametersAction.keepUndefinedParameters`=true to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represents a security breach
Jun 30, 2016 9:41:12 AM hudson.model.ParametersAction filter
WARNING: Skipped parameter `BASE` as it is undefined on `test1`. Set `-Dhudson.model.ParametersAction.keepUndefinedParameters`=true to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represents a security breach
Jun 30, 2016 9:41:12 AM hudson.model.ParametersAction filter
WARNING: Skipped parameter `TARGET` as it is undefined on `test1`. Set `-Dhudson.model.ParametersAction.keepUndefinedParameters
Dhudson.model.ParametersAction.keepUndefinedParameters`=true to allow undefined parameters to be injected as environment variables or `-Dhudson.model.ParametersAction.safeParameters=[comma-separated list]` to whitelist specific parameter names, even though it represents a security breach
{noformat}

I honestly believe is better to explicitly add the needed params than use the hudson.model.ParametersAction.keepUndefinedParameters property